rcl/src/rcl/subscription.c

```diff
@@ -282,6 +282,56 @@ rcl_take(
   return RCL_RET_OK;
 }
 
+rcl_ret_t
+rcl_take_sequence(
+  const rcl_subscription_t * subscription,
+  size_t count,
+  rmw_message_sequence_t * message_sequence,
+  rmw_message_info_sequence_t * message_info_sequence,
+  rmw_subscription_allocation_t * allocation
+)
+{
+  // Set the sizes to zero to indicate that there are no valid messages
+  message_sequence->size = 0;
+  message_info_sequence->size = 0;
+
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Subscription taking n messags");
+  if (!rcl_subscription_is_valid(subscription)) {
+    return RCL_RET_SUBSCRIPTION_INVALID;  // error message already set
+  }
+  RCL_CHECK_ARGUMENT_FOR_NULL(message_sequence, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(message_info_sequence, RCL_RET_INVALID_ARGUMENT);
+
+  if (message_sequence->capacity < count) {
+    RCL_SET_ERROR_MSG("Insufficient message sequence capacity for requested count");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+
+  if (message_info_sequence->capacity < count) {
+    RCL_SET_ERROR_MSG("Insufficient message info sequence capacity for requested count");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+
+  size_t taken = 0;
+  rmw_ret_t ret = rmw_take_sequence(
+    subscription->impl->rmw_handle, count, message_sequence, message_info_sequence, &taken,
+    allocation);
+  if (ret != RMW_RET_OK) {
+    RCL_SET_ERROR_MSG(rmw_get_error_string().str);
+
+    if (RMW_RET_BAD_ALLOC == ret) {
+      return RCL_RET_BAD_ALLOC;
```

Comment:
You might want to consider using this instead:

https://github.com/ros2/rcl/blob/3142616db4b5dac28bf24b266f505e67d9f19897/rcl/src/rcl/common.h#L25-L27